### PR TITLE
142 extended hiccup

### PR DESCRIPTION
**Pull request by hlship**
I've done what I can, you can see I added another issue about how to run tests & etc.

I've tested this in a branch of my project and it works fine as far as I can tell.  It's not something that you use everywhere, but it can be nice (especially with things like Bootstrap, which can deeply nest elements).

If this gets accepted, I have ideas about making it easier to include XML/HTML entities. I sure would like an easy way to &nbsp; , etc.

### CODE REVIEW

File: src/reagent/impl/template.cljs [hunk 1]
```diff
@@ -226,17 +226,51 @@
...
-                 (pr-str v) (comp/comp-name)))
-    (if-some [ne (native-element tag v)]
-      ne
-      (reag-element tag v))))
+  (let [tag  (nth v 0)
+        tags (do
+               (assert (valid-tag? tag)
+                       (str "Invalid Hiccup form: "
+                            (pr-str v) (comp/comp-name)))
+               (if (keyword? tag)
```
Comment (seancorfield) [inline src/reagent/impl/template.cljs]: It seems that if `tag` is not a keyword, then `tags` will be `nil` -- is that intentional? If so, `when` would be better than `if` here.

**Comment by seancorfield** [comment]
It looks like a really nice enhancement -- and a fairly small, localized change! Very nice! I posted a question on one line of the diff.

File: src/reagent/impl/template.cljs (hunk 1 above)
Comment (hlship) [inline src/reagent/impl/template.cljs]: Agreed. I'll update the PR.

**Comment by hlship** [comment]
Thanks. Also, I believe that use of this extended syntax, when appropriate, will generate smaller compiled JavaScript. The compiled JavaScript for creating a nested collection of vectors is larger than the syntax for a single keyword. That leaves a concern about performance, but I don't have the tools to identify if that's a problem (I highly doubt it).

**Comment by pkobrien** [comment]
I agree with Sean. That looks really handy. Nice!

**Pull request merged**